### DISCUSSION TITLE
How does sentry-react-native pull the list of packages in our app?

### DISCUSSION THREAD

**Discussion by fabiendem** [getsentry/sentry-react-native#2807] [Q&A]
Hi,

On Sentry, I can see the list of packages used by our app attached to Errors.
Example:

<img width="540" alt="Screen Shot 2023-02-06 at 15 26 39" src="https://user-images.githubusercontent.com/1186282/216998280-18cdf4c2-9c55-4310-bf3d-c2a7433d6e29.png">

How does Sentry pull this list?
Do we have to add our `package.json` to the bundle (`require("package.json")`)?
Or does it read whatever the libraries expose when bundled in the JS payload?

Thank you

**Comment by krystofwoldrich** [comment]
Hi @fabiendem,
thanks for the question.

This list is created at build time. All top-level modules which were bundled by Metro are recorded.

The [collection script](https://github.com/getsentry/sentry-react-native/blob/7406fdcf9c302cf07305ce618d273e6f033615ec/src/js/tools/ModulesCollector.ts#L21) goes thru bundled files and records from what modules a file is. 

By default, only modules in `${project_root}/node_modules` are recorded.

What do you refer to by `our package.json`?

**Reply by fabiendem** [reply]
Thank you, you've answered my question.

> What do you refer to by our package.json?

I mean the `package.json` of our RN project, the one at the root which declares the dependencies of the project.

**Comment by fabiendem** [comment]
🤔 One feature request in the Sentry dashboard would be to be able to search for Issues based on the version of a given package.

**Reply by krystofwoldrich** [reply]
Thank you for the suggestion.
